packages/block-library/src/list/edit.js

```diff
@@ -13,40 +20,83 @@ export default function ListEdit( {
 	onReplace,
 	className,
 } ) {
-	const { ordered, values } = attributes;
+	const { ordered, values, reversed, start, type } = attributes;
 
 	return (
-		<RichText
-			identifier="values"
-			multiline="li"
-			tagName={ ordered ? 'ol' : 'ul' }
-			onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
-			value={ values }
-			wrapperClassName="block-library-list"
-			className={ className }
-			placeholder={ __( 'Write list…' ) }
-			onMerge={ mergeBlocks }
-			unstableOnSplit={
-				insertBlocksAfter ?
-					( before, after, ...blocks ) => {
-						if ( ! blocks.length ) {
-							blocks.push( createBlock( 'core/paragraph' ) );
-						}
+		<Fragment>
+			<RichText
+				identifier="values"
+				multiline="li"
+				tagName={ ordered ? 'ol' : 'ul' }
+				onChange={ ( nextValues ) => setAttributes( { values: nextValues } ) }
+				value={ values }
+				start={ start }
+				reversed={ reversed }
+				type={ type }
+				wrapperClassName="block-library-list"
+				className={ ordered ? classnames( className, 'ol-type-is-' + type ) : className }
+				placeholder={ __( 'Write list…' ) }
+				onMerge={ mergeBlocks }
+				unstableOnSplit={
+					insertBlocksAfter ?
+						( before, after, ...blocks ) => {
+							if ( ! blocks.length ) {
+								blocks.push( createBlock( 'core/paragraph' ) );
+							}
 
-						if ( after !== '<li></li>' ) {
-							blocks.push( createBlock( 'core/list', {
-								ordered,
-								values: after,
-							} ) );
-						}
+							if ( after !== '<li></li>' ) {
+								blocks.push( createBlock( 'core/list', {
+									ordered,
+									values: after,
+								} ) );
+							}
 
-						setAttributes( { values: before } );
-						insertBlocksAfter( blocks );
-					} :
-					undefined
+							setAttributes( { values: before } );
+							insertBlocksAfter( blocks );
+						} :
+						undefined
+				}
+				onRemove={ () => onReplace( [] ) }
+				onTagNameChange={ ( tag ) => setAttributes( { ordered: tag === 'ol' } ) }
+			/>
+			{
+				ordered &&
+				<InspectorControls>
+					<PanelBody title={ __( 'Ordered List Settings' ) }>
+						<SelectControl
+							label={ __( 'List Type' ) }
+							value={ type }
+							options={ [
+								{ label: 'Decimal', value: '1' },
```

Comment:
Would it be possible to unset the attribute if "Decimal" is selected?

Reply:
Thinking about it more, you are right that it shouldn't be unset. But then there should be one extra option that wouldn't set the attribute. Perhaps we can call this option "Theme", as it would have whatever type the theme defined (usually decimal).

Reply:
@Jackie6 Could you add an extra button then to unset the value?